governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts

```diff
@@ -35,66 +35,89 @@ export const getProposalStatus = (
-          const innerInstructions = governanceAction.instructions
-          innerInstructions.forEach((innerInstruction) => {
-            const multisigParser = MultisigParser.fromCluster(cluster)
-            const parsedInstruction = multisigParser.parseInstruction({
-              programId: innerInstruction.programId,
-              data: innerInstruction.data as Buffer,
-              keys: innerInstruction.keys as AccountMeta[],
-            })
-            acc[parsedInstruction.name] = (acc[parsedInstruction.name] ?? 0) + 1
-          })
-        } else if (governanceAction instanceof PythGovernanceActionImpl) {
-          acc[governanceAction.action] = (acc[governanceAction.action] ?? 0) + 1
-        } else if (governanceAction instanceof SetDataSources) {
-          acc[governanceAction.actionName] =
-            (acc[governanceAction.actionName] ?? 0) + 1
-        } else {
-          acc['unknown'] = (acc['unknown'] ?? 0) + 1
-        }
-      } else {
-        acc[instruction.name] = (acc[instruction.name] ?? 0) + 1
-      }
-      return acc
-    }, {} as Record<string, number>)
+const getInstructionSummariesByName = (
+  parser: MultisigParser,
+  instructions: MultisigInstruction[]
+) =>
+  Object.groupBy(
+    instructions.flatMap((instruction) =>
+      getInstructionSummary(parser, instruction)
+    ),
+    ({ name }) => name
   )
+
+const getInstructionSummary = (
+  parser: MultisigParser,
+  instruction: MultisigInstruction
+) => {
+  if (instruction instanceof WormholeMultisigInstruction) {
+    const { governanceAction } = instruction
+    if (governanceAction instanceof ExecutePostedVaa) {
+      return governanceAction.instructions.map((innerInstruction) =>
+        getTransactionSummary(parser, innerInstruction)
+      )
+    } else if (governanceAction instanceof PythGovernanceActionImpl) {
+      return [{ name: governanceAction.action } as const]
+    } else if (governanceAction instanceof SetDataSources) {
+      return [{ name: governanceAction.actionName } as const]
+    } else {
+      return [{ name: 'unknown' } as const]
+    }
+  } else {
+    return [{ name: instruction.name } as const]
+  }
+}
+
+const getTransactionSummary = (
+  parser: MultisigParser,
+  instruction: TransactionInstruction
+) => {
+  const parsedInstruction = parser.parseInstruction(instruction)
+  switch (parsedInstruction.name) {
+    case 'addPublisher':
+      return {
+        name: 'addPublisher',
+        priceAccount:
+          parsedInstruction.accounts.named['priceAccount'].pubkey.toBase58(),
+        pub: (parsedInstruction.args['pub'] as PublicKey).toBase58(),
+      } as const
+    case 'delPublisher':
+      return {
+        name: 'delPublisher',
+        priceAccount:
+          parsedInstruction.accounts.named['priceAccount'].pubkey.toBase58(),
+        pub: (parsedInstruction.args['pub'] as PublicKey).toBase58(),
+      } as const
+    default:
+      return {
+        name: parsedInstruction.name,
+      } as const
```

Comment:
What is the point of the "as const"

Reply:
In case you're curious, here's a good illustration in the typescript playground of the difference between using the const assertion vs not: https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAJiAyiAtgUygCwJZgOYwC8MAFAIYBOeAXDNBbngJREB8MA3gLABQMdAd2xRgmUpWade-fsDIQ0MAOQAjMgC8ltChgCuFMJxhgy6Wqo1KANHVQYc+WgFYYAXwDc0mXDQAzMroANlDaegZGJmbKEHZYjACigQrWtuhx+IkKtFAUuooeXq68RTy8oJCwCMhpDngA6sKYAMLg0ETiVLT0jCyE7Nx8gsKiHZIDMjByCspqmqFQ+oYcxqZo5rMpMTWMzm4w8pOtUJ6D-D7+QSEwOgvhy5Fr0bG1mWibzwlJjzl5ewfl0BO-BKriAA.

If you take a look at the `.d.ts` tab on the right side, you'll see how typescript infers the types.  If you see how `doSomethingWithConst` gets inferred with literals, it essentially allows any consuming code that matches `name` to `baz` to automatically infer correctly that `something` is 5 and `somethingElse` is `undefined`.